### PR TITLE
Added enum helper functions.

### PR DESCRIPTION
**Pull request by E-B-Smith**
Added BNCProductCategoryAllCategories and BNCCurrencyAllCurrencies functions.

### CODE REVIEW

**Comment by BranchMacMini** [comment]
Result of [Integration 1](https://stlt.herokuapp.com/v1/xcs_deeplink/qabot.stage.branch.io/d336a704184380e6c329924922d2421a/d336a704184380e6c329924922d24e85)
---
*Duration*: 20 minutes and 42 seconds
*Result*: **Build failed 1 test** out of 227

File: Branch-SDK/Branch-SDK/BNCCommerceEvent.m [hunk 1]
```diff
@@ -229,6 +255,203 @@
 BNCCurrency BNCCurrencyZAR = @"ZAR";
 BNCCurrency BNCCurrencyZMW = @"ZMW";
 
+NSArray<BNCCurrency>* BNCCurrencyAllCurrencies(void) {
```
Comment (sojanpr) [inline Branch-SDK/Branch-SDK/BNCCommerceEvent.m]: Should we change this to an enumeration instead?

Comment (E-B-Smith) [reply Branch-SDK/Branch-SDK/BNCCommerceEvent.m]: This is the Swift compatible way to do enums.

Comment (sojanpr) [reply Branch-SDK/Branch-SDK/BNCCommerceEvent.m]: Got it

**Comment by sojanpr** [comment]
LGTM 👍

**Pull request merged**